ui/src/lib/messageHandlers.ts

```diff
@@ -26,6 +26,30 @@ export function extractMessagesFromTasks(tasks: Task[]): Message[] {
   return messages;
 }
 
+// getUsageFromWellknownFields gets token usage from metadata fields they could be defined in (kagent_usage_metadata for kagent agents. usage_metadata, token_usage, usage fields for remote agents)
+// TODO: Should we instead allow for Remote Agent configuration to define where (assuming if) they define their token usage in metadata - defaulting to kagent_usage_metadata.?
+function getUsageFromWellknownFields(metadata: ADKMetadata | undefined): TokenStats | undefined {
+  if (!metadata) {
+    return undefined;
+  }
+
+  const usage = metadata?.kagent_usage_metadata ||
+    metadata?.usage_metadata as ADKMetadata["kagent_usage_metadata"] ||
+    metadata?.token_usage as ADKMetadata["kagent_usage_metadata"] ||
+    metadata?.usage as ADKMetadata["kagent_usage_metadata"] ||
+    undefined;
```

Comment:
this is based on our non-remote stats under `kagent_usage_metadata` and other possible fields that usage would be stored in for remote agents.

we could:
1. Add fields for `remote` type so a user can configure the metadata field(s) for this data (if exposed).
2. Keep as-is
3. [?] Come back to this after figuring out the streaming + Task/history setup.